kf_lib_data_ingest/common/file_retriever.py

```diff
@@ -86,14 +101,30 @@ def _s3_save(protocol, source_loc, dest_obj, auth=None, logger=None):
     raise botocore.exceptions.NoCredentialsError()  # never got the file
 
 
-def _web_save(protocol, source_loc, dest_obj, auth=None, logger=None):
+def _web_save(protocol, source_loc, dest_obj, auth=None, auth_config=None,
+              logger=None):
     """
     Get contents of a file from a web server to a local file-like object.
 
     Preserves the name of the original file if Content-Disposition
     header is provided in the response.
     See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition # noqa E501
 
+    Authentication
+    --------------
+    Currently supports the following authentication schemes:
+        - HTTP Basic Authentication
+        - Token Authentication
+        - OAuth 2 Authentication
+
+    If `auth_config` is provided, inspect the URL to select the authentication
```

Comment:
This design seems very odd to me. If I'm asking for a specific URL, why would I pass in a bunch of different URLs and ask if the first one is among the bunch? I should already know the answer to that. If I know what I'm asking for, and I know what format I need to give you for an auth table, then I should also know how to find the right variables from in the table without passing the buck. I think either the caller should already know what values to use or the caller shouldn't be passing in an auth set at all and it should be loaded statically from a global configuration file _here_ (not at the caller). I think I prefer the latter.

Reply:
Addressed 👍